### DISCUSSION TITLE
Upload images to rack pages

### DISCUSSION THREAD

**Discussion by sfriese** [netbox-community/netbox#15325] [General]
What would be the curl command that will upload an picture to the Images section of a rack page?

URL = https://my.netbox.app/dcim/racks/38/
FILE = myfile.jpg

![image](https://github.com/netbox-community/netbox/assets/122481043/160b0d50-6bc6-426a-90eb-c211e701a214)

**Comment by zmaster7** [comment]
NetBox has a built-in API documentation that can generate a sample curl command for you. You are looking for the /api/extras/image-attachments/ endpoint for uploading images.

https://NetBoxURL/api/schema/swagger-ui/#/extras/extras_image_attachments_create should help point you in the correct direction.

Here is a sanitized curl command output example generated from my instance (Using the multipart form example):

curl -X 'POST' \
  'https://NetBoxURL/api/extras/image-attachments/' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -H 'X-CSRFTOKEN: TOKEN' \
  -F 'content_type=dcim.rack' \
  -F 'object_id=1' \
  -F 'name=myfile' \
  -F 'image=@myfile.jpg;type=image/jpg' \
  -F 'image_height=320' \
  -F 'image_width=220'

**Comment by sfriese** [comment]
Thanks zmaster7, I'll give that a try.
